### PR TITLE
Fix cckbc uri scheme

### PR DESCRIPTION
**Pull request by Xynnn007**
When doing integration test, some bugs occurs and this pr can fix.
As the `reqwest` crate will not automatically try `http` when `https` fails, we must specify the KBS address by `aa_params`. That parameter will be delivered from the gRPC request.

### CODE REVIEW

File: protos/getresource.proto [hunk 1]
```diff
@@ -5,6 +5,7 @@ package getresource;
 message GetResourceRequest {
     string ResourceUri = 1;
     string KbcName = 2;
+    string KbsUri = 3;
```
Comment (sameo) [inline protos/getresource.proto:8]: So what is the `KbsUri` going to look like compared to the `ResourceUri`?

Comment (Xynnn007) [reply protos/getresource.proto:8]: A resource uri is like `kbs://<uri-of-kbs>/<repo>/<resource>/<tag>` following KBS Resource URI for locating the resource. `KbsUri` is like `http://...` or `https://...` which is for pointing a remote KBS. Before we use the scheme `http://` to directly convert a uri `kbs://<uri-of-kbs>/<repo>/<resource>/<tag>` into `http://<uri-of-kbs>`. However, as the distinguishment of `http` and `https`, we need to specify the address of KBS which includes a `http` or a `https` protocol name.

Another choice is to let cc-kbc try to connect with `https` by default. If fails then downgrates to `http`. However `reqwest` crate does not support the automatical downgration, we need to manually implement this.

Comment (sameo) [reply protos/getresource.proto:8]: So `KbsUri` is a URI that contains a scheme and a host path. There's a bit of duplication between the two strings:

`kbs://example.cckbs.org:8081/alice/decryption-key/1` and `https://example.cckbs.org:8081/`.

Why not defining the API as:

```proto3
message GetResourceRequest {
    string ResourcePath = 1; // e.g. /alice/decryption-key/1
    string KbcName = 2; // e.g. cc_kbc
    string KbsUri = 3; // e.g. https://example.cckbs.org:8081/
```

With that approach there is no duplication in the arguments, and we have all the needed information to build fully qualified KBS compliant URIs.

Comment (Xynnn007) [reply protos/getresource.proto:8]: Oh, yes! Great suggestion! Let me fix this

**Review by sameo — APPROVED** [review]
Thanks @Xynnn007 !

**Review by jialez0 — APPROVED** [review]

**Pull request merged**